### ISSUE TITLE
Support for limit(), startsAt(), and endsAt()

### ISSUE THREAD

**Issue by gpbmike** [firebase/emberfire#66]
It would be great to be able to use Firebase's `limit()`, `startsAt()`, and `endsAt()`.

```javascript
store.find('post', { limit: 10, startsAt: 100 });
```

This would require `findQuery` support in emberFire.

**Comment by Emerson** [comment]
Waiting on this as well - hard to go beyond the hello world use case without proper query support

**Comment by jayphelps** [comment]
@Emerson and others, check out https://github.com/firebase/emberfire/pull/169

**Comment by brendanoh** [comment]
I love the serverless pattern and Firebase Hosting is great but without the ability to really send queries I dont see how anyone could build anything substantial using EmberFire.

Let's assume I have an App that needs to do more that just Select by ID and Select All

How do you use this library if you cannot query Firebase?
Is there a way to pass native queries in via the data adapter?

Lets assume I have a user and I want to show all of his Posts but the Firebase repository has more than just his posts stored there. I cannot figure out how to pass in a query param (user = 5) to make that work in a normal way. There doesn't seem to be a way to Sideload either. One thing I can do is if the user has an array of Post Ids in their user model then I can iterate over them and make dozens of API calls into Firebase to get each post but that seems awful.

@katowulf thinks this is "not particularly challenging" but I cannot find any examples and I do not want invent the wheel. How do I 'drop to the Firebase API and use it directly' as I do not see examples, patterns or best practices.

In need of support / guidance / info.

Brendan

**Comment by dustinbrownman** [comment]
@gpbmike This looks great! I'm curious what version of firebase you're using. I get the `undefined is not a function` error whenever I try to call a querying method on `ref` (so `ref.limitToLast(n)` blows up).

**Comment by gpbmike** [comment]
@dustinbrownman you need to be using version 2.

**Comment by codepreneur** [comment]
@gpbmike I tried your solution: https://github.com/webhook/webhook-cms/blob/master/app/adapters/application.js and I get the following error:
```
Error: Cannot set read-only property "contentblocks" on object: <whatnot@model:article::ember392:-JiHiuRBvKtvt12AwJ0A>
    at new Error (native)
    at Error.EmberError (http://localhost:4200/assets/vendor.js:21769:23)
    at Descriptor.computedPropertySet [as _set] (http://localhost:4200/assets/vendor.js:20239:15)
    at Descriptor.computedPropertySetWithSuspend [as set] (http://localhost:4200/assets/vendor.js:20223:14)
    at set (http://localhost:4200/assets/vendor.js:25802:14)
    at __exports__.default.Mixin.create.set (http://localhost:4200/assets/vendor.js:41417:9)
    at http://localhost:4200/assets/whatnot.js:214:15
    at tryCatch (http://localhost:4200/assets/vendor.js:59767:16)
    at invokeCallback (http://localhost:4200/assets/vendor.js:59779:17)
    at publish (http://localhost:4200/assets/vendor.js:59750:11)
```

Any ideas?

**Comment by tstirrat** [comment]
landed in EmberFire 1.4.0+
